web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx

```diff
@@ -295,12 +377,25 @@ export function EnrollEksCluster(props: AgentStepProps) {
         clear={clear}
         disableSelector={fetchClustersAttempt.status === 'processing'}
       />
-      {!hasIamPermError && tableData.currRegion && (
+      {showContent && (
         <>
           <Box mb={2}>
+            <Toggle
+              isToggled={isAutoDiscoveryEnabled}
+              onToggle={() => setAutoDiscoveryEnabled(isEnabled => !isEnabled)}
+            >
+              <Box ml={2} mr={1}>
+                Auto-enroll all EKS clusters for selected region
+              </Box>
+              <ToolTipInfo>
+                Auto-enroll will automatically identify all EKS clusters from
+                the selected region and register them as Kubernetes resources in
+                your infrastructure.
+              </ToolTipInfo>
+            </Toggle>
```

Comment:
i would put the `auto-enroll` toggle after the app discover toggle (so it's closer to the extra note about auto-enrolling)